File: Assignment No_12/Question_2.c
/* Write a program which accept number from user and return the count of Odd Digits.

Input- 2395
Output- 3

Input- 1038
Output- 2

*/
///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

#include<stdio.h>


int  CountOdd(int iNo)
{
	int iDigit=0;
	int iOdd=0;


	if(iNo<0)
	{
	  iNo=-iNo;
	}

	while(iNo!=0)
	{
	  iDigit= iNo %10;

	  if(iDigit % 2 != 0)
	  {
	  	iOdd++;
	  }
	       
         iNo= iNo/10;
	}
	   return iOdd;

}

int main()
{
	int iValue=0, iRet=0; 

	printf("Enter the Number:\n");
	scanf("%d",&iValue);

	iRet=CountOdd(iValue);
	
	printf("Count of Odd Digits  is:%d\n",iRet);

	return 0;
}

File: Assignment No_38/Question_2.java
/* 2.Write a Java program which accept number from user and return the count of odd digits.

Input : 2395
Output : 3

Input : 1018
Output : 2

Input : -1018
Output : 2

Input : 8462
Output : 0
*/
///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

import java.util.*;

class Digits
 {
    private int iNo;

    public Digits(int iNo) 
    {
        this.iNo = iNo;
    }

    public int CountOfOddDigits() 
    {
        int iDigit = 0, iOdd = 0;
        int temp = iNo;

        if (temp < 0) 
        {
            temp = -temp;
        }

        while (temp > 0) 
        {
            iDigit = temp % 10;
            if (iDigit % 2 != 0)
            {
                iOdd++;
            }
            temp = temp / 10;
        }

        return iOdd;
    }
}

class Ass38_2
{
    public static void main(String A[]) 
    {
        Scanner sobj = new Scanner(System.in);

        System.out.println("Enter a number: ");
        int iNo = sobj.nextInt();

        Digits dobj = new Digits(iNo);
        int iRet = dobj.CountOfOddDigits();

        System.out.println("Count of Odd Digits is: " + iRet);
    }
}
